merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/SimulationResults.java

```diff
@@ -55,4 +55,34 @@ public String toString() {
         + ", unfinishedActivities=" + this.unfinishedActivities
         + " }";
   }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    SimulationResults that = (SimulationResults) o;
+
+    if (!startTime.equals(that.startTime)) return false;
+    if (!duration.isEqualTo(that.duration)) return false;
+    if (!realProfiles.equals(that.realProfiles)) return false;
+    if (!discreteProfiles.equals(that.discreteProfiles)) return false;
+    if (!simulatedActivities.equals(that.simulatedActivities)) return false;
+    if (!unfinishedActivities.equals(that.unfinishedActivities)) return false;
+    if (!topics.equals(that.topics)) return false;
+    return events.equals(that.events);
+  }
+
+  @Override
+  public int hashCode() {
+    int result = startTime.hashCode();
+    result = 31 * result + duration.hashCode();
+    result = 31 * result + realProfiles.hashCode();
+    result = 31 * result + discreteProfiles.hashCode();
+    result = 31 * result + simulatedActivities.hashCode();
+    result = 31 * result + unfinishedActivities.hashCode();
+    result = 31 * result + topics.hashCode();
+    result = 31 * result + events.hashCode();
+    return result;
+  }
```

Comment:
Quoting a book quoted on a stackoverflow discussion:

> from Joshua Bloch, Effective Java, Chapter 3, Item 9
> 
>  The value 31 was chosen because it is an odd prime. If it were even and the multiplication overflowed, information would be lost, as multiplication by 2 is equivalent to shifting. The advantage of using a prime is less clear, but it is traditional. A nice property of 31 is that the multiplication can be replaced by a shift and a subtraction for better performance: 31 * i == (i << 5) - i. Modern VMs do this sort of optimization automatically.